platforms/core-configuration/model-core/src/main/java/org/gradle/internal/snapshot/impl/IsolatedArray.java

```diff
@@ -57,7 +58,7 @@ public Object[] isolate() {
     @Nullable
     @Override
     public <S> S coerce(Class<S> type) {
-        return null;
+        return Cast.uncheckedCast(elements);
```

Comment:
This seems unsafe. We need to make sure to that the thing we're returning is assignable to type